proxy/src/telemetry/metrics/labels.rs

```diff
@@ -48,8 +50,11 @@ enum Direction {
     Outbound,
 }
 
-#[derive(Clone, Debug, Hash, Eq, PartialEq)]
-pub struct DstLabels(Arc<str>);
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct DstLabels {
+    formatted: Arc<str>,
+    original: Arc<HashMap<String, String>>,
```

Comment:
`IndexMap`?

Reply:
Something about storing both the formatted and original labels _seems_ wrong to me, but I can't think of any better way to represent this, so 👍

Reply:
agree, why not derive `formatted` from `original` when needed? is this an optimization?

Reply:
this has to be a HashMap so that it can be handed directly to protobuf

Reply:
@olix0r oh okay nevermind then!
@siggy IIRC the early formatting was an optimization for metrics & I think we probably want to preserve that...

Reply:
Yes, as @hawkw mentioned, this is an optimization so that we can amortize the cost of serialization. Note that this pattern isn't all that onerous in Rust, since `formatted` and `original` are explicitly _immutable_ -- it's literally impossible to change the contents of that map once DstLabels is created.  And, furthermore, because DstLabels does not provide mutable access to these fields, we have reasonable constraints/guarantees that they can't get out of sync at runtime!